File: week17/week17-2.cpp
//week17-2.cpp LeetCode 學習計畫 66. Plus One
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int N=digits.size();
        int carry=1;
        for(int i=N-1;i>=0;i--){
            digits[i]=digits[i]+carry;
            if(digits[i]>=10){//大於等於10,進位
                carry=1;
                digits[i]=digits[i]%10;
            }else{
                carry=0;
            }
        }
        if(carry==0) return digits;//沒進位,直接當答案
        else{
            vector<int> ans(N+1); //多1位(進位的carry)
            ans[0]=carry;//進位的放最前面
            for(int i=0;i<N;i++){
                ans[i+1]=digits[i];//左邊要避開carry的那一位
            }
            return ans;
        }
        
    }

File: week17/week17-2.py
#week17-2.py LeetCode 學習計畫 66. Plus One
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        #python
        N=len(digits) #有幾位數
        carry=1
        for  i in range(N-1,-1,-1): #到過來的迴圈
            digits[i]=digits[i]+carry#加1
            if digits[i]>=10:#如果加完>=10有進
                carry=1
                digits[i]=digits[i]%10#餘數留原地
            else:#沒有進位
                carry=0
        if carry==0:
            return digits
        else:
            return [carry]+digits